cpp/CMakePresets.json

```diff
@@ -11,6 +11,7 @@
       "hidden": true,
       "generator": "Ninja",
       "cacheVariables": {
+        "CMAKE_EXPORT_COMPILE_COMMANDS": "ON",
```

Comment:
It's less than a megabyte on my machine; I think it's not expensive and it's worthwhile to enable clang tools by default

Reply:
I agree that the generated file is not huge, but I also think these presets should enable just what's necessary for development. They serve as documentation and starting points for users. It's easy to enable additional options on top of an existing preset, AFAIK.

Reply:
But which tools need this in CI? I use this file on my machine for LSP on my editor.

Reply:
The presets aren't just for CI builds, as @pitrou says it's a starting point for users. I would say this is an easy and useful default to provide for users- myself included. I'll move this to a different issue, though, since it's not directly pertinent to offset storage

Reply:
https://github.com/apache/arrow/issues/36124